core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4223,6 +4223,17 @@ private void analyzeColumnMask(String currentIdentity, Table table, QualifiedObj
             analysis.addColumnMask(table, column, expression);
         }
 
+        private Identity getIdentityForExpression(ViewExpression expression)
+        {
+            // TODO: the ViewExpression we get from access control should include the full Identity instead of just the user name
+            if (expression.getIdentity().equals(session.getUser())) {
+                return session.getIdentity();
```

Comment:
This is not a good assumption and a potential escalation of privileges. The original user associated with the filter/mask may not have permissions that are inherited from the enabled roles in the current session's identity.

Reply:
Mask or filter is a form of a view and this exactly what we are doing in for views in `io.trino.sql.analyzer.StatementAnalyzer.Visitor#analyzeView`.

Notice that there is also opposite problem where the original user associated with the filter/mask may have more enabled roles that in the current session's identity. We don't know nothing here about roles as we get only user name. I assume that we should enable all roles for expression user.

The same problem exists for views, and we are trying to fix it with:  https://github.com/trinodb/trino/pull/13991

I think the solution for both cases should be either:
 - view owner and mask or filter owner should be defined as a full identity (that requires a check if given roles are still applicable to given user)
 - or we should always enable all roles that user (system and catalog roles)

Reply:
The point is, I think, that we don't know what privileges the original user had. We cannot use an `Identity` with no roles here, because it has exactly the opposite problem - it may have no privileges at all.

Also, it does not change much from the status quo, because currently a mask or filter will "re-use" the first `Identity` for this user that was added earlier in the analysis. Which is most likely the current session's identity.

Reply:
@martint any update on this?